### ISSUE TITLE
Casting in fields' reader may be unnecessary

### ISSUE THREAD

**Issue by duskmoon314** [rust-embedded/svd2rust#671]
I noticed this warning in the generated code:

```rust
warning: casting to the same type is unnecessary (`u32` -> `u32`)
   --> d1-pac/src/ccu/pll_ddr_pat0_ctrl.rs:265:25
    |
265 |         WAVE_BOT_R::new((self.bits & 0x0001_ffff) as u32)
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `(self.bits & 0x0001_ffff)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_cast
    = note: `#[warn(clippy::unnecessary_cast)]` on by default
```

There are many similar warnings. I haven't checked whether it is simple to tell that `RegisterSpec::Ux` and `#fty` are the same types. Maybe add `#[allow(...)]` if not.
